lmfdb/elliptic_curves/web_ec.py

```diff
@@ -168,6 +168,32 @@ def latex_equation(ainvs):
                     '{:+}'.format(a6) if a6 else '',
                     r'\)'])
 
+def homogeneous_latex_equation(ainvs):
+    a1,a2,a3,a4,a6 = [int(a) for a in ainvs]
+    return ''.join([r'\(y^2z',
+                    '+xyz' if a1 else '',
+                    '+yz^2' if a3 else '',
+                    '=x^3',
+                    '+x^2z' if a2==1 else '-x^2z' if a2==-1 else '',
+                    '{:+}xz^2'.format(a4) if abs(a4)>1 else '+xz^2' if a4==1 else '-xz^2' if a4==-1 else '',
+                    '{:+}z^3'.format(a6) if abs(a6)>1 else '+z^3' if a6==1 else '-z^3' if a6==-1 else '',
+                    r'\)'])
+
+def short_latex_equation(ainvs):
+    a1,a2,a3,a4,a6 = [ZZ(a) for a in ainvs]
+    c4 = -27*a1^4 - 216*a1^2*a2 + 648*a1*a3 - 432*a2^2 + 1296*a4
+    c6 = 54*a1^6 + 648*a1^4*a2 - 1944*a1^3*a3 + 2592*a1^2*a2^2 - 3888*a1^2*a4 - 7776*a1*a2*a3 + 3456*a2^3 - 15552*a2*a4 + 11664*a3^2 + 46656*a6
+    for p in c4.gcd(c6).prime_divisors():
+        while c4.valuation(p) >= 4 and c6.valuation(p) >= 6:
+            c4 = c4.divide_knowing_divisible_by(p**4)
+            c6 = c6.divide_knowing_divisible_by(p**6)
```

Comment:
c4 //= p**4 is surely fast enough, but this is fine too.

Reply:
I generally try to avoid using / or // in python (who knows, the semantics could change in python4).